R/board_azure.R

```diff
@@ -20,31 +26,59 @@
 #' \dontrun{
 #' # To create a board that you can write to, you'll need to supply one
 #' # of `key`, `token`, or `sas` to AzureStor::blob_container()
-#' container <- AzureStor::blob_container(url, key = "my-key")
-#' board <- board_azure(container)
+#' # First, we create a board using the classic Azure blob API
+#' url <- "https://myaccount.blob.core.windows.net/mycontainer"
+#' container <- AzureStor::blob_container(url, sas = "my-sas")
+#' board <- board_azure(container, "path/to/board")
 #' board %>% pin_write(iris)
+#'
+#' # ADLSgen2 is a modern, efficient way to access blobs
+#' # - Use 'dfs' instead of 'blob' in the account URL to use the ADLSgen2 API
+#' # - We reuse the board created via the blob API above
+#' adls_url <- "https://myaccount.dfs.core.windows.net/mycontainer"
+#' container <- AzureStor::storage_container(adls_url, sas = "my-sas")
+#' board <- board_azure(container, "path/to/board")
+#' board %>% pin_list()
+#' board %>% pin_read("iris")
 #' }
-board_azure <- function(container, n_processes = 10, versioned = TRUE, cache = NULL) {
+board_azure <- function(container, path = "", n_processes = 10, versioned = TRUE, cache = NULL) {
   check_installed("AzureStor")
 
-  cache <- cache %||% board_cache_path(paste0("azure-", hash(container$endpoint$url)))
+  if (path == "/") {
+    path <- ""
+  }
+  board_path <- fs::path(container$endpoint$url, container$name, path)
+  cache <- cache %||% board_cache_path(paste0("azure-", hash(board_path)))
+  if (path != "") {
+    if(inherits(container, "file_share")) {
+      try(AzureStor::create_storage_dir(container, path, recursive = TRUE), silent = TRUE)
```

Comment:
Why the try?

Reply:
The different storage services all behave in annoyingly different ways. For file storage, creating a dir that already exists is an error; the other services will silently ignore it.

I could probably do a better job of providing a consistency layer in AzureStor, but I don't think there's many people who want to do the exact same operations on all 3 services, like here.